scripts/mensa.coffee

```diff
@@ -16,26 +16,87 @@
 #   kiliankoe
 
 
+default_mensa = process.env.HUBOT_DEFAULT_MENSA or "Alte Mensa"
+
+
+mensen = [
+    names: ["zelt", "zeltmensa"]
+    id: 79
+  ,
+    names: ["alte", "alte mensa"]
+    id: 80
+  ,
+    names: ["reichenbachstraße"]
+    id: 81
+  ,
+    names: ["mensologie"]
+    id: 82
+  ,
+    names: ["siedepunkt"]
+    id: 83
+  ,
+    names: ["tharandt"]
+    id: 84
+  ,
+    names: ["palucca"]
+    id: 85
+  ,
+    names: ["wu", "wundtstraße"]
+    id: 86
+  ,
+    names: ["stimm-gabel", "stimmgabel"]
+    id: 87
+  ,
+    names: ["johannstadt"]
+    id: 88
+  ,
+    names: ["u-boot", "uboot"]
+    id: 89
+  ,
+    names: ["zittau"]
+    id: 90
+  ,
+    names: ["sport"]
+    id: 91
+  ,
+    names: ["kreuzgymnasium", "kreuz"]
+    id: 92
+  ]
+
+
+mappedMensa = mensen.reduce((map, mensa) ->
+    mensa.names.forEach (name) ->
+      map[name] = mensa.id
+    map
+  , {})
+
+
 module.exports = (robot) ->
+
+  generic_resp_func = (mensa, callback) ->
+    mensaKey = mensa.toLowerCase()
+    if not mappedMensa.hasOwnProperty mensaKey
+      callback "Kenne leider keine solche Mensa..."
+    else
+      getMeals(mensa, mappedMensa[mensaKey], callback)
+
+  getMeals = (name, mensa, callback) ->
+    robot.http("http://openmensa.org/api/v2/canteens/#{mensa}/days/today/meals")
+      .get() (err, res, body) ->
+        data = JSON.parse body
+        callback "Heute @ *#{name}*:\n#{data.map(formatOutput).join('\n')}"
+
+
   robot.respond /mensa$/i, (msg) ->
-    getMeals("Alte Mensa", 79, robot, msg)
+    generic_resp_func(default_mensa, (m) -> msg.send m)
 
-  robot.respond /mensa (.*)/i, (msg) ->
+  robot.respond /mensa (\S.*)/i, (msg) ->
     mensa = msg.match[1]
-    index = mensen.indexOf(mensa.toLowerCase()) + 79
-    if index == 78
-      msg.send "Kenne leider keine solche Mensa..."
-      return
-    getMeals(mensa, index, robot, msg)
+    generic_resp_func(mensa, (m) -> msg.send m)
 
   robot.respond /mensen/i, (msg) ->
     msg.send "Ich kann dir heutige Speisepläne für die folgenden Mensen holen:\n - #{mensen.join('\n - ')}\nSprich' mich einfach mit `matthias mensa <mensa>` an."
```

Comment:
This just lists a whole bunch of `[object Object]`, you're going to want to make this list the new names of the objects.